### PR TITLE
Change enrollment code redemption to the receipt page.

### PR DESCRIPTION
**Pull request by vkaracic**
Reopening this PR: https://github.com/edx/ecommerce/pull/963
Only change is the increase from three to five checks for order completion. https://github.com/edx/ecommerce/compare/vkaracic/SOL-2073?expand=1#diff-c797ef6707ae3dfb1ee8828592244a9eR170

SOL-2073

### CODE REVIEW

**Comment by edx-webhook** [comment]
Thanks for the pull request, @vkaracic! It looks like you're a member of a company that does contract work for edX. If you're doing this work as part of a paid contract with edX, you should talk to edX about who will review this pull request. If this work is not part of a paid contract with edX, then you should ensure that there is an OSPR issue to track this work in [JIRA](https://openedx.atlassian.net), so that we don't lose track of your pull request.

<a   href="https://openedx-webhooks.herokuapp.com/github/process_pr?number=1105&repo=edx%2Fecommerce"   rel="nofollow" >Create an OSPR issue for this pull request</a>.

File: ecommerce/extensions/order/models.py [hunk 1]
```diff
@@ -21,6 +21,9 @@ def contains_coupon(self):
         return any(line.product.get_product_class().name == 'Coupon' for line in
                    self.basket.all_lines())
 
+    class Meta(object):
+        get_latest_by = 'date_placed'
```
Comment (mjfrey) [inline ecommerce/extensions/order/models.py]: Why this change?

Comment (vkaracic) [reply ecommerce/extensions/order/models.py]: I can't remember, probably because of the test. I'll revert it back.

**Review by mjfrey — APPROVED** [review]
Squash commits.
